### ISSUE TITLE
GridDecoration or VerticalAxisDecoration verticalValues truncated

### ISSUE THREAD

**Issue by AMuellerAtAHS** [infinum/flutter-charts#20]
Hi,

I try to use this package with a growing chart, which works and renders really well. It is really simple to implement.

But if I want to show the values of the vertical decoration (x-axis labels), the values will be truncated if the number indexes and the size of the values are growing.

Here is a code snippet based on the flutter example project.

```
import 'dart:async';

import 'package:charts_painter/chart.dart';
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({Key? key, required this.title}) : super(key: key);

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  int _counter = 0;
  Timer? _timer;
  final List<double> _dataPoints = [0];
  final bool _showBars = true;

  @override
  void initState() {
    super.initState();
    _timer =
        Timer.periodic(const Duration(milliseconds: 300), _updateDataPoints);
  }

  @override
  void dispose() {
    _timer!.cancel();
    super.dispose();
  }

  void _incrementCounter() {
    setState(() {
      _counter++;
    });
  }

  void _updateDataPoints(Timer timer) {
    _recordData();
  }

  void _recordData() {
    setState(() {
      _dataPoints.add(_counter.toDouble());
    });
  }

  double _calcVerticalSteps() {
    final division = (_dataPoints.length / 10);
    if (division < 1) return 1;
    if (division < 10) return 10;
    if (division < 100) return 100;
    return 1000;
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
      ),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            const Text(
              'You have pushed the button this many times:',
            ),
            Text(
              '$_counter',
              style: Theme.of(context).textTheme.headline4,
            ),
            Container(
              padding: const EdgeInsets.only(top: 20),
              child: Chart(
                state: ChartState(
                  ChartData.fromList(
                    _dataPoints
                        .map((e) => BubbleValue<void>(e.toDouble()))
                        .toList(),
                  ),
                  itemOptions: const BubbleItemOptions(
                    padding: EdgeInsets.symmetric(horizontal: 2.0),
                    color: Colors.red,
                    maxBarWidth: 1.0,
                  ),
                  backgroundDecorations: [
                    GridDecoration(
                      showVerticalGrid: true,
                      verticalAxisStep: _calcVerticalSteps(),
                      showVerticalValues: true,
                      verticalTextAlign: TextAlign.center,
                      textStyle: const TextStyle(color: Colors.red),
                      horizontalAxisStep: 1,
                      gridColor: Theme.of(context).dividerColor,
                      showHorizontalValues: true,
                      verticalValuesPadding: EdgeInsets.zero,
                    ),
                    SparkLineDecoration(
                      lineWidth: 2.0,
                      lineColor: Theme.of(context).colorScheme.primary,
                    ),
                  ],
                  foregroundDecorations: [
                    ValueDecoration(
                      alignment: _showBars
                          ? Alignment.bottomCenter
                          : const Alignment(0.0, -1.0),
                      textStyle: Theme.of(context).textTheme.button!.copyWith(
                          color: (_showBars
                                  ? Theme.of(context).colorScheme.onPrimary
                                  : Theme.of(context).colorScheme.primary)
                              .withOpacity(0.0)),
                    ),
                  ],
                ),
              ),
            ),
          ],
        ),
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: _incrementCounter,
        tooltip: 'Increment',
        child: const Icon(Icons.add),
      ), // This trailing comma makes auto-formatting nicer for build methods.
    );
  }
}
```
Maybe someone could help here, how I could change my code, to show correct (not truncated values).

**Comment by AMuellerAtAHS** [comment]
Perhaps I was not clear. 
I just want to calculate the vertical steps, but without truncating them.
Is it possible to prevent the titles from truncating? Or give them a abbreviation?

**Comment by lukaknezic** [comment]
render-object is now merged into master and is part of `2.0.0`
